src/Build/Instance/ProjectInstance.cs

```diff
@@ -1358,22 +1383,23 @@ ICollection<ProjectItemInstance> IItemProvider<ProjectItemInstance>.GetItems(str
 
         /// <summary>
         /// Record an import opened during evaluation.
-        /// Does nothing: not needed for project instances.
         /// </summary>
         void IEvaluatorData<ProjectPropertyInstance, ProjectItemInstance, ProjectMetadataInstance, ProjectItemDefinitionInstance>.RecordImport(
             ProjectImportElement importElement,
             ProjectRootElement import,
             int versionEvaluated,
             SdkResult sdkResult)
         {
+            _imports.Add(import.FullPath);
+            ((IEvaluatorData<ProjectPropertyInstance, ProjectItemInstance, ProjectMetadataInstance, ProjectItemDefinitionInstance>)this).RecordImportWithDuplicates(importElement, import, versionEvaluated);
         }
 
         /// <summary>
         /// Record an import opened during evaluation. Include duplicates
-        /// Does nothing: not needed for project instances.
         /// </summary>
         void IEvaluatorData<ProjectPropertyInstance, ProjectItemInstance, ProjectMetadataInstance, ProjectItemDefinitionInstance>.RecordImportWithDuplicates(ProjectImportElement importElement, ProjectRootElement import, int versionEvaluated)
         {
+            _importsIncludingDuplicates.Add(import.FullPath);
```

Comment:
These methods don't seem like they're ever used?

Reply:
Also, if either of these is called, it would mean `Imports` and `ImportsWithDuplicates` would be out-of-date relative to `_imports` and `_importsIncludingDuplicates`.

Reply:
They're used during evaluation, when any of the overloads which call Initialize are used (not the one used when creating a ProjectInstance from a Project)

Reply:
Regarding out-of-dateness, the AsReadOnly is a wrapper around the collection. It's not a snapshot copy of the collection.